sematic/db/migrations/20230706150252_pre_alembic_parity.py

```diff
@@ -0,0 +1,339 @@
+                    updated_at timestamp WITH time zone NOT NULL,
+                    started_at timestamp,
+                    ended_at timestamp,
+                    resolved_at timestamp,
+                    failed_at timestamp,
+                    parent_id character(32),
+                    description TEXT,
+                    tags TEXT NOT NULL,
+                    source_code TEXT NOT NULL,
+                    root_id character(32) NOT NULL,
+                    nested_future_id character(32),
+                    resource_requirements_json JSONB,
+                    exception_metadata_json JSONB,
+                    container_image_uri TEXT,
+                    external_exception_metadata_json JSONB,
+                    original_run_id character(32),
+                    cache_key TEXT,
+                    user_id character(32) REFERENCES users(id),
+
+                    PRIMARY KEY (id),
+
+                    FOREIGN KEY(root_id) REFERENCES runs (id)
+                );
+                """
+            )
+
+            conn.execute("CREATE INDEX ix_runs_cache_key ON runs_new (cache_key);")
+            conn.execute(
+                "CREATE INDEX ix_runs_function_path ON runs_new (function_path);"
+            )
+
+            conn.execute(
+                """
+                INSERT INTO runs_new
+                    SELECT
+                        id,
+                        future_state,
+                        name,
+                        function_path,
+                        created_at,
+                        updated_at,
+                        started_at,
+                        ended_at,
+                        resolved_at,
+                        failed_at,
+                        parent_id,
+                        description,
+                        tags,
+                        source_code,
+                        root_id,
+                        nested_future_id,
+                        resource_requirements_json,
+                        exception_metadata_json,
+                        container_image_uri,
+                        external_exception_metadata_json,
+                        original_run_id,
+                        cache_key,
+                        user_id
+                    FROM runs;
+                """
+            )
+
+            conn.execute("DROP TABLE runs;")
+            conn.execute("ALTER TABLE runs_new RENAME TO runs;")
+
+            conn.execute(
+                """
+                CREATE TABLE artifacts_new (
+                    id character(40) NOT NULL,
+                    json_summary JSONB NOT NULL,
+                    created_at timestamp NOT NULL,
+                    updated_at timestamp NOT NULL,
+                    type_serialization JSONB NOT NULL,
+
+                    PRIMARY KEY (id)
+                );
+                """
+            )
+            conn.execute("INSERT INTO artifacts_new SELECT * FROM artifacts;")
```

Comment:
Please always explicitly specify the columns to select in all these queries, as previous noop downgrades due to #302 have resulted in extra garbage columns in devs' and in users' dbs:

```
[...]
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) table artifacts_new has 5 columns but 6 values were supplied
[SQL: INSERT INTO artifacts_new SELECT * FROM artifacts;]
```

Reply:
done